pkg/scaffold/internal/filesystem/mock.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filesystem
+
+import (
+	"bytes"
+	"io"
+)
+
+// mockFileSystem implements FileSystem
+type mockFileSystem struct {
```

Comment:
Why we need the mockFileSystem? Why not create a FileSystem with mock data to do the tests?

Reply:
`FileSystem` is not an struct with data we can mock. It is an abstraction of the actual file system, of the disk space and the files and directories that are created. In https://github.com/kubernetes-sigs/kubebuilder/pull/1339#discussion_r374089278 you can find the explanation why we should not use the real `FileSystem` to test the `machinery` package. Lets keep the discussion in that thread.